### PR TITLE
LVIIICNET-171 Member fee

### PR DESCRIPTION
**Pull request by bohdansydor**

### CODE REVIEW

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  cancel() {
+    this.memberFee = new MemberFee();
+    this.router.navigate(['admin', 'fee']);
+  }
+  add() {
+    this.cancel();
+  }
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts]: empty line

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.less [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+body {
+  background-color: #3e94ec;
+  font-size: 16px;
+  font-weight: 400;
+  text-rendering: optimizeLegibility;
+}
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.less]: empty lines

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.spec.ts]: empty lines

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/member-fee.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  styleUrls: ['./member-fee.component.less']
+})
+export class MemberFeeComponent implements OnInit {
+  
+  constructor(private router: Router) { }
+
+  ngOnInit() {
+    
+  }
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/member-fee.component.ts]: empty lines

File: TheraLang.Web/Controllers/MemberFeeController.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    [ApiController]
+    public class MemberFeeController : ControllerBase
+    {
+        private readonly IMemberFeeService _memberFeeService;
+
+
+        public MemberFeeController(IMemberFeeService memberFeeService)
+        {
+            _memberFeeService = memberFeeService;
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/Controllers/MemberFeeController.cs]: empty line

File: TheraLang.Web/ViewModels/MemberFeeViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+using System.Threading.Tasks;
+
+namespace TheraLang.Web.ViewModels
+{
+    public class MemberFeeViewModel
+    {
+        public int Id { get; set; }
+        public DateTime FeeDate { get; set; }
+        public decimal FeeAmount { get; set; }
+
```
Comment (oleksiistetsiouk) [inline TheraLang.Web/ViewModels/MemberFeeViewModel.cs]: empty lines

File: TheraLang.Web/Controllers/MemberFeeController.cs (hunk 5 above)
Comment (bohdansydor) [reply TheraLang.Web/Controllers/MemberFeeController.cs]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit, ChangeDetectorRef } from '@angular/core';
+import { MemberFee } from 'src/app/shared/models/member-fee/member-fee';
+import { MemberFeeService } from 'src/app/core/http/manager/fee.service';
+import { Router } from '@angular/router';
+
```
[the remark this answers is not in the document]
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts]: Done

File: TheraLang.BLL/Interfaces/IMemberFeeService.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace TheraLang.BLL.Interfaces
+{
+    public interface IMemberFeeService
+    {
+        Task<IEnumerable<MemberFeeDto>> GetMemberFeesAsync();
+        Task AddAsync(MemberFeeDto memberFeeModel);
+        Task DeleteAsync(int id);
+    }
+
```
[the remark this answers is not in the document]
Comment (bohdansydor) [reply TheraLang.BLL/Interfaces/IMemberFeeService.cs]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        <input matInput [matDatepicker]="picker" [(ngModel)]="memberFee.feeDate" placeholder="Choose a date">
+        <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+        <mat-datepicker #picker></mat-datepicker>
+    </div>
+    <div class="form-group">
+
+        <label><b>FeeAmount</b></label>
+
+        <input type="number" [(ngModel)]="memberFee.feeAmount" class="form-control" />
+
```
[the remark this answers is not in the document]
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.html]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts (hunk 1 above)
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/create-fee/create-fee.component.ts]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.less (hunk 2 above)
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/get-fee/get-fee.component.less]: Done

File: TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/member-fee.component.ts (hunk 4 above)
Comment (bohdansydor) [reply TheraLang.Web/ClientApp/src/app/modules/manager/member-fee/member-fee.component.ts]: Done

File: TheraLang.Web/ViewModels/MemberFeeViewModel.cs (hunk 6 above)
Comment (bohdansydor) [reply TheraLang.Web/ViewModels/MemberFeeViewModel.cs]: Done

File: TheraLang.BLL/Services/MemberFeeService.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+namespace TheraLang.BLL.Services
+{
+   public class MemberFeeService: IMemberFeeService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        public MemberFeeService(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+        public async Task<IEnumerable<MemberFeeDto>> GetMemberFeesAsync()
```
Comment (mod1fy) [inline TheraLang.BLL/Services/MemberFeeService.cs]: Add lines between methods

File: TheraLang.DAL/Configuration/MemberFeeConfiguratoin.cs [hunk 11]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TheraLang.DAL.Entities;
+
+namespace TheraLang.DAL.Configuration
+{
+    class MemberFeeConfiguratoin: IEntityTypeConfiguration<MemberFee>
```
Comment (Dreylak) [inline TheraLang.DAL/Configuration/MemberFeeConfiguratoin.cs]: Rename to Configuration

File: TheraLang.BLL/Services/MemberFeeService.cs (hunk 10 above)
Comment (bohdansydor) [reply TheraLang.BLL/Services/MemberFeeService.cs]: Done))))

**Pull request merged**